### PR TITLE
feat(cli): added option on the new command to create a deployment project or a flow project

### LINKED ISSUE #5755: Add option on the Jina CLI to allow deployment centric projects by adding a `deployment` flag
**Describe the feature**
Add option for `jina new` to be either flow centric or deployment centric

**Your proposal**
Add a flag to the `jina new` command to allow optional creation of deployment centric

### PR DESCRIPTION
**Pull request by asuzukosi**
I added a feature to allow users to be able to create a deployment themed project from the `new` command on the cli. This would allow users to specify an optional `--type` flag which would allow them to create a project as either a `deployment` or a `flow`. If no `--type` is specified it defaults to a `flow`. If a wrong `--type` is specified it still defaults to a `flow` This will ensure backwards compatibility with current users of the project. Examples of correct commands are:
- `jina new hello-world`
- `jina new hello-world --type flow`
- `jina new hello-world --type deployment`
- `jina new hello-world --type`

For the deployment project type, it still used the yaml format that was used in the `flow` type. It creates a `deployment.yml` file.

**Goals:**

- resolves #5755
- Allows more versatility for project creation
- Provides more information that can be used on the documentation
- Retains backwards compatibility

**Future work. **
- Provide information about this feature on the documentation
- Add option to create `flow` or `deployment` in either pythonic or YAMLish syntax

### CODE REVIEW

**Comment by asuzukosi** [comment]
Hello @JoanFM  I have implemented the feature we spoke about earlier today on issue #5755 , please I would appreciate if you could help me review it. Thank you.

**Review by JoanFM — CHANGES_REQUESTED** [review]
It looks very good to me. We only need to make sure to mention it in Documentation

**Comment by asuzukosi** [comment]
@JoanFM, thank you for the feedback. Please how can I add it to the documentation, is there any required format? I would need some assistance with that

**Comment by JoanFM** [comment]
Hey @asuzukosi,

You may need to edit the Create First Project in the Documentation. These changes have to be made in the markdown files under `docs`. Just be aware that the capacity to serve a single Deployment via jcloud is not yet possible.

**Comment by asuzukosi** [comment]
@JoanFM, thanks I'll get to it.

**Comment by asuzukosi** [comment]
@JoanFM, I've updated the documentation to reflect the new feature. Please take a look at it when you are available.

File: docs/get-started/create-app.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
 # {fas}`folder-plus` Create First Project
 
-Let's build a toy application with Jina. To start, use Jina CLI to make a new project:
+Let's build a toy application with Jina. To start, use Jina CLI to make a new project, we can either make a new project as a deployment or a flow:
```
Comment (alexcg1) [inline docs/get-started/create-app.md:3]:
```suggestion
Let's build a toy application with Jina. To start, use Jina CLI to make a new Deployment or a Flow: 
```
Please capitalize "Deployment" and "Flow" throughout.

File: docs/get-started/create-app.md [hunk 2]
```diff
@@ -1,13 +1,20 @@
 # {fas}`folder-plus` Create First Project
 
-Let's build a toy application with Jina. To start, use Jina CLI to make a new project:
+Let's build a toy application with Jina. To start, use Jina CLI to make a new project, we can either make a new project as a deployment or a flow: 
 
+To create a flow project with Jina CLI
```
Comment (alexcg1) [inline docs/get-started/create-app.md:5]:
```suggestion
To create a flow project with Jina CLI:
```
Please use colon in this kind of sentence before code blocks

File: docs/get-started/create-app.md [hunk 3]
```diff
@@ -17,21 +24,37 @@ hello-jina/
...
+hello-jina/
+    |- client.py
+    |- deployment.yml
+    |- executor1/
+            |- config.yml
+            |- executor.py
+```
+
+
+- `flow.yml/deployment.yml` is the configuration file for the {class}`~jina.Flow` or `~jina.Deployment` based on what type of project you create.
```
Comment (alexcg1) [inline docs/get-started/create-app.md:38]:
```suggestion
- `flow.yml/deployment.yml` is the configuration file for the {class}`~jina.Flow` or {class}`~jina.Deployment` based on what type of project you create.
```

File: docs/get-started/create-app.md [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
+```
+
+To create a deployment project with Jina CLI
+```bash
+jina new hello-jina --type=deployment
 ```
 
 This creates a new project folder called `hello-jina` with the following file structure:
 
+For a flow project:
```
Comment (alexcg1) [inline docs/get-started/create-app.md:17]: Would be good to group Flow/Deployment file structures with tabs. Then we don't need "For a Flow project", etc.

Here's an example of the syntax (GH won't let me suggest directly via usual tool, and bc of all the backticks I can't type it directly and have it render):

image

File: docs/get-started/create-app.md [hunk 5]
```diff
@@ -131,7 +154,7 @@ tensor([[[0.9594, 0.9373],
 
 ## Deploy to JCloud
 
-JCloud offers free CPU and GPU instances to host Jina project. Let's deploy your first project to JCloud:
+JCloud offers free CPU and GPU instances to host Jina project. Let's deploy your first project to JCloud (Note that JCloud is only available for flow projects):
```
Comment (alexcg1) [inline docs/get-started/create-app.md:157]:
```suggestion
\`\`\`admonition
:class: important

At present, JCloud is only available for Flows. We are currently working on supporting Deployments.
\`\`\`

JCloud offers free CPU and GPU instances to host your Jina project. Let's deploy your first project:
```

GH won't let me put my _actual_ suggestion in (it won't render), so I've had to put backslashes in front of the backticks. My actual suggestion doesn't include the backticks (`\`)

**Review by alexcg1 — CHANGES_REQUESTED** [review]
Excellent PR! Thanks so much! Just a few small notes. Please let me know if you have any questions regarding my comments. I've had to butcher them a bit because GitHub won't actually render them properly (since my suggested code changes include backticks and GH's markdown rendering engine gets very confused)

**Comment by asuzukosi** [comment]
@JoanFM @alexcg1  I have made the updates to the documentation to correct all the mistakes noted by @alexcg1  and I have also used the tab system to differentiate between Flow and Deployment in the documentation. Please review when you are available, Thank you.

File: docs/get-started/create-app.md [hunk 6]
```diff
@@ -1,34 +1,175 @@
...
+At present, JCloud is only available for Flows. We are currently working on supporting Deployments.
+```
+
+You've just finished your first toy Jina project, congratulations! You can now start your own project.
+
+````
+
+````{tab} Flow
+
+To create a flow project with Jina CLI:
```
Comment (alexcg1) [inline docs/get-started/create-app.md:147]: "Flow" should have capital first letter. Also "Deployment" - can you check throughout the doc that they're capitalized correctly?

```suggestion
To create a Flow project with Jina CLI:
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/get-started/create-app.md (hunk 6 above)
Comment (asuzukosi) [reply docs/get-started/create-app.md:147]: Alright, Thank you. Will get to that @alexcg1

**Comment by asuzukosi** [comment]
Hello @alexcg1, I have made the corrections. I also noticed in my last ci/cd test my code was failing an integration test, but I didn't work on any integration aspect of the project, please what should I do about this?

File: docs/get-started/create-app.md [hunk 7]
```diff
@@ -1,34 +1,175 @@
 # {fas}`folder-plus` Create First Project
 
-Let's build a toy application with Jina. To start, use Jina CLI to make a new project:
+Let's build a toy application with Jina. To start, use Jina CLI to make a new Deployment or a Flow: 
 
+````{tab} Deployment
+
+To create a Deployment project with Jina CLI
```
Comment (alexcg1) [inline docs/get-started/create-app.md:7]: Can remove lines like this. They're redundant

File: docs/get-started/create-app.md [hunk 8]
```diff
@@ -1,34 +1,175 @@
...
+At present, JCloud is only available for Flows. We are currently working on supporting Deployments.
+```
+
+You've just finished your first toy Jina project, congratulations! You can now start your own project.
+
+````
+
+````{tab} Flow
+
+To create a Flow project with Jina CLI:
```
Comment (alexcg1) [inline docs/get-started/create-app.md:147]: Kill this line too

```suggestion
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/get-started/create-app.md (hunk 8 above)
Comment (asuzukosi) [reply docs/get-started/create-app.md:147]: Alright, I'll get to It

**Comment by asuzukosi** [comment]
Hello @alexcg1 , I've removed the redundant lines in the documentation. Thank you for the feedback

File: docs/get-started/create-app.md [hunk 9]
```diff
@@ -131,7 +269,7 @@ tensor([[[0.9594, 0.9373],
 
 ## Deploy to JCloud
 
-JCloud offers free CPU and GPU instances to host Jina project. Let's deploy your first project to JCloud:
+JCloud offers free CPU and GPU instances to host Jina project. Let's deploy your first project to JCloud (Note that JCloud is only available for Flow projects):
```
Comment (alexcg1) [inline docs/get-started/create-app.md:272]: As mentioned before, I think we should use [admonition](https://myst-parser.readthedocs.io/en/v0.16.1/syntax/syntax.html#directives-a-block-level-extension-point) instead

**Review by alexcg1 — CHANGES_REQUESTED** [review]
Just need that one admonition in there and I think we're good to go :)

File: docs/get-started/create-app.md (hunk 9 above)
Comment (asuzukosi) [reply docs/get-started/create-app.md:272]: @alexcg1 ,  Thanks a lot! don't know how I missed that haha

**Comment by asuzukosi** [comment]
@alexcg1 , I have made the final suggested changes to the documentation. Thank you very much for the support thus far, this was fun :) CC @JoanFM

**Review by alexcg1 — APPROVED** [review]
LGTM :+1: Fantastic work!

**Review by JoanFM — APPROVED** [review]
Thanks for this awesome contribution @asuzukosi

**Pull request merged**